File: testdata/jennies/builders/constant_assignment/PythonBuilder/builders/sandbox.py
import typing
from ..cog import builder as cogbuilder
from ..models import sandbox


class SomeStruct(cogbuilder.Builder[sandbox.SomeStruct]):
    _internal: sandbox.SomeStruct

    def __init__(self):
        self._internal = sandbox.SomeStruct()

    def build(self) -> sandbox.SomeStruct:
        """
        Builds the object.
        """
        return self._internal    
    
    def editable(self) -> typing.Self:    
        self._internal.editable = True
    
        return self
    
    def readonly(self) -> typing.Self:    
        self._internal.editable = False
    
        return self
    
    def auto_refresh(self) -> typing.Self:    
        self._internal.auto_refresh = True
    
        return self
    
    def no_auto_refresh(self) -> typing.Self:    
        self._internal.auto_refresh = False
    
        return self
    

File: testdata/jennies/builders/constant_assignment/RustBuilder/src/builders/sandbox.rs
use crate::cog;
use crate::types::sandbox;

#[derive(Debug, Clone)]
pub struct SomeStructBuilder {
    internal: sandbox::SomeStruct,
    pub(crate) errors: Vec<cog::BuildError>,
}

impl SomeStructBuilder {
    pub fn new() -> Self {
        Self {
            internal: sandbox::SomeStruct::default(),
            errors: Vec::new(),
        }
    }
}

impl Default for SomeStructBuilder {
    fn default() -> Self {
        Self::new()
    }
}

impl SomeStructBuilder {
    pub fn editable(mut self) -> Self {
        self.internal.editable = true;

        self
    }
}

impl SomeStructBuilder {
    pub fn readonly(mut self) -> Self {
        self.internal.editable = false;

        self
    }
}

impl SomeStructBuilder {
    pub fn auto_refresh(mut self) -> Self {
        self.internal.auto_refresh = Some(true);

        self
    }
}

impl SomeStructBuilder {
    pub fn no_auto_refresh(mut self) -> Self {
        self.internal.auto_refresh = Some(false);

        self
    }
}

impl cog::Builder<sandbox::SomeStruct> for SomeStructBuilder {
    fn build(&self) -> Result<sandbox::SomeStruct, Vec<cog::BuildError>> {
        if !self.errors.is_empty() {
            return Err(self.errors.clone());
        }

        Ok(self.internal.clone())
    }
}
